### PR TITLE
Only show ios7 error message when attempting to use access control

### PR DESCRIPTION
**Pull request by tupakapoor**
As discussed in comments on 71784e3

### CODE REVIEW

File: SimpleKeychain/A0SimpleKeychain.m [hunk 1]
```diff
@@ -356,12 +356,12 @@ - (NSDictionary *)queryFetchOneByKey:(NSString *)key message:(NSString *)message
                                       (__bridge id)kSecAttrAccount: key,
                                       }];
 #if TARGET_OS_IPHONE
-    if (self.useAccessControl && floor(NSFoundationVersionNumber) > NSFoundationVersionNumber_iOS_7_1) {
-        if (message) {
+    if (self.useAccessControl) {
+        if (message && floor(NSFoundationVersionNumber) > NSFoundationVersionNumber_iOS_7_1) {
             query[(__bridge id)kSecUseOperationPrompt] = message;
+        } else {
+            NSLog(@"Error trying to access to non available kSecUseOperationPrompt in iOS7");
```
Comment (hzalaz) [inline SimpleKeychain/A0SimpleKeychain.m]: Let's remove the log and I'll merge it

Comment (tupakapoor) [inline SimpleKeychain/A0SimpleKeychain.m]: If you want to remove the log message then I'll go back to the same format as before and just remove the log message - this was just so it didn't print a misleading error message when `useAccessControl` was false. Let me know what you'd like, thanks!

Comment (hzalaz) [inline SimpleKeychain/A0SimpleKeychain.m]: Yeah, the thing is that is not a "real error" more like a warning, let's keep it but changing it a bit
```objc
#ifdef DEBUG
else {
    NSLog(@"Warning: Ignoring use of kSecUseOperationPrompt that is not available in iOS 7");
}
#endif
```

On a side note, I'll research if I can mark some methods/properties as disabled for specific iOS version so the compiler will complain instead.

Comment (grasmeyer) [inline SimpleKeychain/A0SimpleKeychain.m]: The problem is that when self.useAccessControl == NO and the iOS version is greater than 7.1, the log message always appears when this method is called.

You should fix this by breaking the first if statement into 2 nested if statements:

    if (self.useAccessControl) {
        if (floor(NSFoundationVersionNumber) > NSFoundationVersionNumber_iOS_7_1) {
            if (message) {
                query[(__bridge id)kSecUseOperationPrompt] = message;
            }
        } else {
            NSLog(@"Error trying to access to non available kSecUseOperationPrompt in iOS7");
        }
    }

Comment (hzalaz) [inline SimpleKeychain/A0SimpleKeychain.m]: @grasmeyer the change in the PR won't print anything unless accessControl is `YES`.
@tupakapoor I am merging this as it is, no point wasting time for something for such an older iOS version. Thanks for the PR.

**Pull request merged**